consensus/broadcast/delayedBroadcast.go

```diff
@@ -245,37 +251,50 @@ func (dbb *delayedBlockBroadcaster) SetValidatorData(broadcastData *shared.Delay
 	return nil
 }
 
-// SetFinalConsensusMessageForValidator sets the consensus message to be broadcast by validator when its turn comes
-func (dbb *delayedBlockBroadcaster) SetFinalConsensusMessageForValidator(message *consensus.Message, consensusIndex int) error {
-	if message == nil {
-		return spos.ErrNilConsensusMessage
+// SetFinalProofForValidator sets the header proof to be broadcast by validator when its turn comes
+func (dbb *delayedBlockBroadcaster) SetFinalProofForValidator(
```

Comment:
Where is the previous final message handled?
before equivalent proofs activation?

Reply:
these methods were added on purpose for equivalent proofs